src/Umbraco.Web/Extensions/UdiExtensions.cs

```diff
@@ -36,8 +36,30 @@ public static IPublishedContent ToPublishedContent(this Udi udi)
                         return umbracoHelper.TypedMember(memberAttempt.Result);
                     break;
             }
-            
+
             return null;
         }
+
+        /// <summary>
+        /// An extension method to easily acquire the integer Id for a given udi
+        /// </summary>
+        /// <param name="udi"></param>
+        /// <returns>An <see cref="int"/> identifier if the item is found, -1 otherwise</returns>
+        public static int ToIntId(this Udi udi)
```

Comment:
You would end up with 

```
var url = library.NiceUrl(Services.EntityService.GetIdForUdi(myUdi))
```

Since ServiceContext is exposed on all base classes. A GetIdForUdi method can be added to UmbracoHelper and then wrap this EntityService method. Then you could do

```
Umbraco.GetIdForUdi(...)
````

And if you wanted to get a Url by Guid then we need overloads for the `UmbracoHelper.Url(Udi udi)`